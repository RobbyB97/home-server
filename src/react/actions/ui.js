export const navToggle = () => ({
    type: 'NAV_TOGGLE'
})


export const navOff = () => ({
    type: "NAV_OFF"
})