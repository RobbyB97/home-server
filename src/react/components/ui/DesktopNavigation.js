/* Libraries */
import React from 'react'


/* Component */
export const DesktopNavigation = () => (
    <div id="DesktopNavigation">
        <ul>
            <li>
                <a href="#about">about</a>
            </li>
            <li>
                <a href="#services">services</a>
            </li>
            <li>
                <a href="#skills">skills</a>
            </li>
            <li>
                <a href="#experience">experience</a>
            </li>
            <li>
                <a href="#projects">projects</a>
            </li>
        </ul>
    </div>
)


export default DesktopNavigation