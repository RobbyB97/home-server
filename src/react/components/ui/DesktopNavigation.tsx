export {}

/* Libraries */
import * as React from 'react'


/* Component */
export class DesktopNavigation extends React.Component<{}, {}> {
    public render() {
        return (
            <div id="DesktopNavigation">
                <ul>
                    <li>
                        <a href="#about">about</a>
                    </li>
                    <li>
                        <a href="#services">services</a>
                    </li>
                    <li>
                        <a href="#skills">skills</a>
                    </li>
                    <li>
                        <a href="#experience">experience</a>
                    </li>
                    <li>
                        <a href="#projects">projects</a>
                    </li>
                </ul>
            </div>
        )
    }
}


export default DesktopNavigation;