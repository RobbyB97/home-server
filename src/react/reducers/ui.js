const UIDefaultState = {
    nav: false,
    white: false
}


export default (state = UIDefaultState, action) => {
    switch(action.type) {
        case 'NAV_TOGGLE':
            return {
                ...state,
                nav: state.nav ? false : true
            }

        case 'NAV_OFF':
            return {
                ...state,
                nav: false
            }

        case 'WHITE_ON':
            return {
                ...state,
                white: true
            }

        case 'WHITE_OFF':
            return {
                ...state,
                white: false
            }

        default:
            return state
    }
}