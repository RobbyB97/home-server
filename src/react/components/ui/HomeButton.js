/* Libraries */
import React from 'react';
import {NavLink} from 'react-router-dom';
import {connect} from 'react-redux';

import {navToggle} from '../../actions/ui';


/* Component */
export class HomeButton extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            show: false
        };
    }

    componentDidMount() {
        const component = this;
        document.addEventListener('scroll', () => {
            component.toggle();
        });
    }

    toggle = () => {
        if (window.pageYOffset > 300) {
            this.setState({
                show: true
            });
        } else {
            this.setState({
                show: false
            });
        }
    }

    navToggle = () => {
        this.props.navToggle();
    }

    render() {
        return (
            <section id="HomeButton" 
                data-active={this.props.nav || this.props.white}
                onClick={this.props.nav && this.navToggle}
                >
                
                {this.props.internal ?
                    <a  
                        href="#app" 
                        className="homeButton__logo"
                        data-show={this.state.show}
                    >
                        B
                    </a>
                :
                    <NavLink 
                        to="/" 
                        onClick={this.toTop}
                        className="homeButton__logo"
                        data-show={this.state.show}
                    >
                        B
				    </NavLink>
                }
            </section>
        );
    }
}


/* Connect to store */
const mapStateToProps = (state) => {
    return {
        nav: state.ui.nav,
        white: state.ui.white
    }
};

const mapDispatchToProps = (dispatch) => ({
    navToggle: () => {
        dispatch(navToggle());
    }
});


/* Export */
export default connect(mapStateToProps, mapDispatchToProps)(HomeButton);