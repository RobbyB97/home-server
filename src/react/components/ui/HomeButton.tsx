export {};

/* Libraries */
import * as React from 'react';
import {NavLink} from 'react-router-dom';
import {connect} from 'react-redux';

import {navToggle} from '../../actions/ui';


/* Component */
interface Props {
    nav?: boolean,
    internal?: boolean,
    navToggle?: typeof navToggle
}
interface State {
    show: boolean,
    nav: boolean,
    navToggle: typeof navToggle
}


export class HomeButton extends React.Component<Props, State> {
    constructor(props: any) {
        super(props);
        this.state = {
            ...this.state,
            show: false,
            nav: false
        }
    }

    componentDidMount() {
        document.addEventListener('scroll', () => {
            this.toggle();
        })
    }

    toggle = () => {
        if (window.pageYOffset > 300) {
            this.setState({
                show: true
            });
        } else {
            this.setState({
                show: false
            });
        }
    }

    navToggle = () => {
        this.state.navToggle();
    }

    public render() {
        const {nav} = this.state;
        const {internal} = this.props;
        const {show} = this.state;
        const {navToggle} = this.state;

        return (
            <div id="HomeButton" 
                className={nav ? "nav":""}
                onClick={nav && navToggle}
                >
                
                {internal ?
                    <a href="#app" className={show ?
                        "homeButton__logo show":
                        "homeButton__logo"
                    }>B</a>
                :
                    <NavLink to="/" className={show ?
                        "homeButton__logo show":
                        "homeButton__logo"}>
                        B
				    </NavLink>
                }
            </div>
        );
    }
}


/* Connect to store */
const mapStateToProps = (state): any => {
    return {
        nav: state.ui.nav
    };
};

const mapDispatchToProps = (dispatch): any => ({
    navToggle: () => {
        dispatch(navToggle());
    }
});


/* Export */
export default connect(mapStateToProps, mapDispatchToProps)(HomeButton);