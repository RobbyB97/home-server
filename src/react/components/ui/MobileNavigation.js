import React from 'react'
import {connect} from 'react-redux'

import {navToggle} from '../../actions/ui'


export class MobileNavigation extends React.Component {
    constructor(props) {
        super(props);

        this.navToggle = this.navToggle.bind(this);
        this.body = document.getElementsByTagName("BODY")[0];
        this.html = document.getElementsByTagName("HTML")[0];
    }

    navToggle = () => {
        // Reset scroll position
        if (this.props.nav) {   // Turning nav off
            this.body.style.overflow = "unset";
            this.html.style.overflow = "unset";
            this.body.style.overflowX = "hidden";

        } else {    // Turning nav on
            this.body.style.overflow = "hidden";
            this.html.style.overflow = "hidden";
        }

        this.props.navToggle();
    }

    render() {
        return (
            <section id="MobileNavigation">
                <div className="mobileNavigation__button"
                    data-nav={this.props.nav}
                >
                    <div onClick={this.navToggle}>
					    {this.props.nav ? '✕':'☰'}
                    </div>
				</div>

                <nav className="mobileNavigation__menu"
                    data-nav={this.props.nav}
                >
                    <ul className="mobileNavigation__links">
                        <li>
                            <a href="#About" onClick={this.navToggle}>About</a>
                        </li>
                        
                        <li>
                            <a href="#Skills" onClick={this.navToggle}>Skills</a>
                        </li>

                        <li>
                            <a href="#Projects" onClick={this.navToggle}>Projects</a>
                        </li>

                        <li>
                            <a href="#Experience" onClick={this.navToggle}>Experience</a>
                        </li>
                    </ul>
                </nav>
            </section>
        );
    }
}


const mapStateToProps = (state) => {
    return {
        nav: state.ui.nav,
        white: state.ui.white
    }
};

const mapDispatchToProps = (dispatch) => ({
    navToggle: () => {
        dispatch(navToggle());
    }
});


export default connect(mapStateToProps, mapDispatchToProps)(MobileNavigation);