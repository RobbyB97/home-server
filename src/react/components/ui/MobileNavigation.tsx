export {}

/* Libraries */
import * as React from 'react';
import {connect} from 'react-redux';

import {navToggle} from '../../actions/ui';


/* Component */
interface Props {
    nav: boolean,
    navToggle: typeof navToggle
}
interface State {
    nav: boolean,
    navToggle: typeof navToggle
}

export class MobileNavigation extends React.Component<Props, State> {
    constructor(props: any) {
        super(props);
        this.state = {
            ...this.state
        }
    }

    navToggle = () => {
        this.state.navToggle();
    };

    public render() {
        const {nav} = this.state;
        const {navToggle} = this.state;

        return (
            <div id="MobileNavigation">
                <div className={nav ?
                    "mobileNavigation__button nav":
                    "mobileNavigation__button"}>
                    <div onClick={navToggle}>
					    {nav ? '✕':'☰'}
                    </div>
				</div>

                <div className={nav ? 
                    "mobileNavigation__menu nav":
                    "mobileNavigation__menu"}>
                    <ul className="mobileNavigation__links">
                        <li>
                            <a href="#about" onClick={navToggle}>about</a>
                        </li>
                        <li>
                            <a href="#services" onClick={navToggle}>services</a>
                        </li>
                        <li>
                            <a href="#skills" onClick={navToggle}>skills</a>
                        </li>
                        <li>
                            <a href="#experience"onClick={navToggle}>experience</a>
                        </li>
                        <li>
                            <a href="#projects" onClick={navToggle}>projects</a>
                        </li>
                    </ul>
                </div>
            </div>
        );
    }
}

/* Connect to store */
const mapStateToProps = (state) => ({
    nav: state.ui.nav,
});

const mapDispatchToProps = (dispatch) => ({
    navToggle: () => {
        dispatch(navToggle())
    }
});


/* Export */
export default connect(mapStateToProps, mapDispatchToProps)(MobileNavigation);