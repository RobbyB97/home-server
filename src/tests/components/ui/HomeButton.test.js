import React from 'react'
import {shallow} from 'enzyme'

import {HomeButton} from '../../../react/components/ui/HomeButton'


test('Render HomeButton (external)', () => {
    const wrapper = shallow(<HomeButton />)
    expect(wrapper).toMatchSnapshot()
})


test('Render HomeButton (internal)', () => {
    const wrapper = shallow(<HomeButton internal />)
    expect(wrapper).toMatchSnapshot()
})