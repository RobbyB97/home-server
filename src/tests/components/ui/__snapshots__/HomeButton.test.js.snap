// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render HomeButton (external) 1`] = `
<section
  id="HomeButton"
>
  <NavLink
    activeClassName="active"
    ariaCurrent="true"
    className="homeButton__logo"
    to="/"
  >
    B
  </NavLink>
</section>
`;

exports[`Render HomeButton (internal) 1`] = `
<section
  id="HomeButton"
>
  <a
    className="homeButton__logo"
    href="#app"
  >
    B
  </a>
</section>
`;
